text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,397 @@
+authors to only those categories that have been pre-selected by the crates.io
+team.
+
+- **Clarity:** In addition to having a list of recommending keywords to choose
+from, providing descriptions alongside those keywords helps users decide which
+crates should be labeled with those keywords too. The same applies to categories
+as well.
+
+- **Endorsement:** As discussed above, the introduction of a category is a very
+heavy decision since it can be seen as an endorsement from the Rust project
+itself. If keywords are promoted primarily based on usage data then this
+reflects less on the project and more on its users, reducing this pressure on
+the crates.io team.
+
+- **Grassroots:** When keywords are promoted to being recommended based
+primarily on usage data then there is less perception of "top-down
+decision-making" compared to the fixed list of categories "dictated" by the
+crates.io team.
+
+- **Choice:** Choosing from the list of all available keywords can quickly lead
+to choice overload. Having recommended keywords can counter this by giving
+crate authors a short list of keywords to choose from, and only if they don't
+find any fitting keywords they can drop down to the full list of keywords or
+introduce a new one.
+
+- **Bike-shedding:** As with previous points in this list, promoting keywords
+based on usage data means that any bike-shedding discussions on what keywords to
+use have already happened by users, and the crates.io team only has to partake
+in the logistics of adding information about existing keywords.
+
+
+# Flat vs. Nested
+
+**Flat** means that there is no hierarchy of terms. A crate related to doing
+HTTP requests could for example use the keywords `web`, `http`, and
+`http-client`. 
+
+**Nested** means that there is a hierarchy of terms. The same crate could be
+tagged with `web/http-client`, where `web` is the parent category and
+`http-client` the child.
+
+**Comparison:**
+
+- **Simplicity:** Having a flat namespace is generally simpler for crate authors
+and users to understand.
+
+- **Duplicates:** Nested hierarchies work well when using a closed set of terms.
+When users can create new terms without moderation the chance of having
+duplicate terms increases significantly though since both the parent and child
+terms could now be duplicated. One hypothetical example of that being
+`maths` and `science/maths` terms being used.
+
+- **Refinement:** Nested hierarchies make it possible to refine the search space
+from e.g. `science` to `science/biology`. In a flat namespace you could
+potentially combine multiple keywords to achieve similar results. Displaying a
+list of [related keywords](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265600285)
+to the one the user is currently looking at can help with this though.
+
+- **Clarity:** In a flat namespace the chance of a single keyword being used for
+multiple purposes is slightly increased. With nested hierarchies, the parent
```

Comment:
Personally, this reads as having a lot of what look like opinions that bias the reader to the opinion of the author.

Reply:
I'm not quite sure what you mean here. It is difficult to be truly neutral in the analysis (especially since the RFC is in favour of a particular solution) but I do think that if we can change the wording to be more neutral, that's a good idea.